collective/searchandreplace/searchreplaceutility.py

```diff
@@ -353,13 +364,12 @@ def getTextFields(obj):
                     text_fields.append(field)
                     continue
                 # ITextLine inherits from IText.
-                # We want to replace in texts, but not textlines.
-                # Maybe this can be made configurable.
-                if ITextLine.providedBy(field):
+                # That we want to replace in texts, but not textlines
+                # is by configuration.
+                if not include_textline_fields and ITextLine.providedBy(field):
                     continue
-                if not IText.providedBy(field):
-                    continue
-                text_fields.append(field)
+                if IText.providedBy(field):
+                    text_fields.append(field)
```

Comment:
The conditions in the dexterity code are checked in a different order than in the archetypes code. They seem to do the same thing, so should be no problem, but for clarity it may be best to keep the order the same. Or put this in a separate function or method.

Reply:
Will change order.